apps/examples/linux_rpc_demo/linux_rpc_demo.c

```diff
@@ -0,0 +1,599 @@
+/*
+ * Copyright (c) 2021, L&T Technology Services Ltd.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*This is a sample demonstration application that showcases usage of proxy
+ *from the remote core.
+ *This application is meant to run on the remote CPU running linux.
+ *This application can print to the master console and perform file I/O through
+ *rpmsg channels.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <openamp/open_amp.h>
+#include <openamp/rpmsg_rpc_client_server.h>
+#include "rsc_table.h"
+#include "platform_info.h"
+#include "linux-rpmsg-rpc-demo.h"
+
+#define REDEF_O_CREAT   0000100
+#define REDEF_O_EXCL    0000200
+#define REDEF_O_RDONLY  0000000
+#define REDEF_O_WRONLY  0000001
+#define REDEF_O_RDWR    0000002
+#define REDEF_O_APPEND  0002000
+#define REDEF_O_ACCMODE 0000003
+
+#define LPRINTF(format, ...) printf(format, ##__VA_ARGS__)
+#define LPERROR(format, ...) LPRINTF("ERROR: " format, ##__VA_ARGS__)
+
+static struct rpmsg_rpc_clt *rpmsg_default_rpc;
+static int fd, bytes_written, bytes_read;
+static struct polling poll;
+static atomic_int nacked;
```

Comment:
This name is misleading. "NACK" is "negative acknowledgment", saying "no" to something. Here it's a flag saying that the other end has processed the request.